### DISCUSSION TITLE
The files for foundry FundMe are too much and confusing

### DISCUSSION THREAD

**Discussion by solidity-cobra** [Cyfrin/foundry-full-course-cu#2191] [General]
please are all this files necessary? or  i am just to go through it, the files includes the interaction, integration. i dont really understand what they do

**Comment by solidity-cobra** [comment]
@engrpips

**Reply by solidity-cobra** [reply]
@EngrPips i need your help

**Reply by EngrPips** [reply]
Hello @solidity-cobra, Here I come. As @thebraveman17 has mentioned, the files may seem overwhelming at first, but they will make more sense once you step through then and make sense of what they do. Take it one at a time.

**Reply by solidity-cobra** [reply]
should i move on hoping to see their use later on or i have to understand the content?
it gets confusing as he modifies them over and over again

**Reply by EngrPips** [reply]
If you cannot grasp it at once, you can proceed and get back to it later.

**Reply by solidity-cobra** [reply]
alright sir, i will just try to understand it before moving so i dont get things more difficult, thanks for your time. i really appreciate you for coming through for me at anytime.

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it!

**Comment by urosognjenovic** [comment]
The amount of files might seem overwhelming and unnecessary at first. However, if you go one step at a time, everything will make perfect sense.

**Reply by solidity-cobra** [reply]
thanks sir, i really appreciate

**Comment by solidity-cobra** [comment]
Hello @EngrPips I'm done with foundry FundMe, what would be your advice, should I continue to the next phase or what.... please enlighten me as I want to be guided.
I need to get it right

**Reply by solidity-cobra** [reply]
@thebraveman17  I will also love to hear what you have to say

**Reply by EngrPips** [reply]
> Hello @EngrPips I'm done with foundry FundMe, what would be your advice, should I continue to the next phase or what.... please enlighten me as I want to be guided. I need to get it right

You can continue the course or maybe build a project based on what you have learned so far.

**Reply by solidity-cobra** [reply]
Alright sir
Thank you

**Reply by EngrPips** [reply]
You are welcome, Friend. My best advice to you is don't stop and keep learning. Keep growing, and keep applying what you learn. If you rest, you will rust.

**Reply by urosognjenovic** [reply]
Just keep going, it will get easier after you finish a couple of projects.

**Reply by solidity-cobra** [reply]
My terminal is lunching with root.
How can I stop this 
![Uploading 17240850147806526602400814804761.jpg…]()

**Reply by EngrPips** [reply]
Hello @solidity-cobra, Your image is not loading for me.

**Reply by solidity-cobra** [reply]
![IMG_20240819_114057](https://github.com/user-attachments/assets/6a1ec006-7e6f-48c3-8edb-217cd34f9610)

**Comment by solidity-cobra** [comment]
![IMG_20240819_114057](https://github.com/user-attachments/assets/b589f481-8e27-41a0-9ecb-3ff5da7b475a)

**Reply by EngrPips** [reply]
I am unsure how to help here, lol. Does it not usually open from the root before?

**Reply by solidity-cobra** [reply]
No sir

**Reply by EngrPips** [reply]
I haven't used `wsl` before, so I might not be helpful here. @cromewar, we need your help.

**Reply by solidity-cobra** [reply]
@cromewar

**Comment by Alam-Sol** [comment]
@EngrPips Hello 
Can you tell me the purpose of Interactions file. What is the actual use of it?

**Reply by EngrPips** [reply]
Hello @Alam-Sol, The primary purpose of interactions file is to interact with an instance of our deploy contract through the command line. Please ask questions if this does not make sense to you.

**Reply by Alam-Sol** [reply]
Thank you for your answer @EngrPips  but i am still confused :( 
We already deploy our contract through deployfundme. Why we need interactions file to deploy again ?

**Reply by EngrPips** [reply]
Do we deploy `FundMe`  in any of the interaction files? Please point out the line where you think we deployed `FundMe` contract in any of the interaction files.

**Comment by Alam-Sol** [comment]
Yes, you're correct—we're not deploying the contract. So, what's the purpose of this? Could you explain a bit more? I know it's a basic question, but I'd appreciate some clarity.

**Reply by EngrPips** [reply]
The purpose of the interactions script is to interact with an already deployed instance of our contract. Interacting here means calling functions on our contract to change state. You can easily tell what an interaction script does by its name, and that is why you will see a script named `fundFundMe` which means it sends funds to a deployed instance of our `FundMe` contract. Honestly, you can retake the section if you are not able to make sense of everything and it will make more sense on your second pass.

**Reply by Alam-Sol** [reply]
I rewatch the whole section and now you words make sense. Thank you :)

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
